src/porepy/applications/boundary_conditions/model_boundary_conditions.py

```diff
@@ -290,3 +291,366 @@ def bc_values_displacement(self, bg: pp.BoundaryGrid) -> np.ndarray:
+        """Boundary condition type for mechanics.
+
+        Neumann boundary conditions are defined on all boundaries.
+
+        Parameters:
+            sd: Subdomain for which to define boundary conditions.
+
+        Returns:
+            bc: Boundary condition object.
+
+        """
+        domain_sides = self.domain_boundary_sides(sd)
+        bc = pp.BoundaryConditionVectorial(sd, domain_sides.all_bf, "neu")
+        if sd.dim < self.nd:
+            # No displacement is implemented on grids of co-dimension >= 1.
+            return bc
+        bc.internal_to_dirichlet(sd)
+        faces_to_fix = self.faces_to_fix(sd)
+        # Fix y and z displacements on face 1 and 2, x and z displacements on face 3.
+        dir = [
+            np.array([False, True, True]),  # Fix y and z on face 1 (west)
+            np.array([False, True, True]),  # Fix y and z on face 2 (east)
+            np.array([True, False, True]),  # Fix x and z on face 3 (south)
+        ]
+        for i, face in enumerate(faces_to_fix):
+            bc.is_dir[:, face] = dir[i]
+            bc.is_neu[:, face] = ~dir[i]  # Negate for Neumann
+        return bc
+
+    def faces_to_fix(self, sd: pp.Grid) -> list[np.int64]:
+        """Return list of faces to fix to avoid rigid body motions.
+
+        See class documentation for more details on the choice of points.
+
+        Parameters:
+            sd: Subdomain for which to define boundary conditions.
+
+        Returns:
+            List of face indices to fix, ordered as in the class documentation.
+
+        """
+        domain_sides = self.domain_boundary_sides(sd)
+        box = self.domain.bounding_box
+
+        # Point 1 is on the center top of the west boundary, having min x coordinate and
+        # mean y coordinate.
+        x_mean = 0.5 * (box["xmax"] + box["xmin"])
+        y_mean = 0.5 * (box["ymax"] + box["ymin"])
+        z_max = box["zmax"]
+        point_1 = np.array([box["xmin"], y_mean, z_max])
+        pts = sd.face_centers[:, domain_sides.west]
+        ind_1 = domain_sides.west.nonzero()[0][
+            np.argmin(pp.distances.point_pointset(point_1, pts))
+        ]
+        # Point 2 is on the center top of the east boundary.
+        point_2 = np.array([box["xmax"], y_mean, z_max])
+        pts = sd.face_centers[:, domain_sides.east]
+        ind_2 = domain_sides.east.nonzero()[0][
+            np.argmin(pp.distances.point_pointset(point_2, pts))
+        ]
+        # Point 3 is on the center top of the south boundary, having min y coordinate
+        # and mean x coordinate.
+        point_3 = np.array([x_mean, box["ymin"], z_max])
+        pts = sd.face_centers[:, domain_sides.south]
+        ind_3 = domain_sides.south.nonzero()[0][
+            np.argmin(pp.distances.point_pointset(point_3, pts))
+        ]
+        return [ind_1, ind_2, ind_3]
+
+
+class GravityMagnitude:
+    """Mixin class for gravity magnitude computation for different materials.
+
```

Comment:
```suggestion
The density is taken as the reference density for the material.

```
'Reference density' is not an ideal term, improvements are most welcome.